src/components/ColorInput/__test__/colorInput.spec.js

```diff
@@ -69,13 +69,78 @@ describe('<ColorInput />', () => {
         expect(changeFn).toHaveBeenCalledWith({ hex: '#000000', alpha: null });
     });
 
+    it('should call onChange with alpha 1 when type an alpha greater than 100', () => {
+        const changeFn = jest.fn();
+        const component = mount(<ColorInput onChange={changeFn} />);
+        component.find('input[type="number"]').simulate('change', { target: { value: '150' } });
+        expect(changeFn).toHaveBeenCalledWith({ hex: '#000000', alpha: 1 });
+    });
+
+    it('should call onChange with alpha 0 when type an alpha smaller than 0', () => {
+        const changeFn = jest.fn();
+        const component = mount(<ColorInput onChange={changeFn} />);
+        component.find('input[type="number"]').simulate('change', { target: { value: '-10' } });
+        expect(changeFn).toHaveBeenCalledWith({ hex: '#000000', alpha: 0 });
+    });
+
     it('should call onClick callback', () => {
         const clickFn = jest.fn();
         const component = mount(<ColorInput onClick={clickFn} />);
         component.find('input[type="text"]').simulate('click');
         expect(clickFn).toHaveBeenCalled();
     });
 
+    it('should call onFocus with the current value', () => {
+        const focusFn = jest.fn();
+        const value = { hex: '#ccc', alpha: 1, isValid: true };
+        const component = mount(<ColorInput value={value} onFocus={focusFn} />);
+        component.find('input[type="text"]').simulate('focus');
+        expect(focusFn).toHaveBeenCalledWith(value);
+    });
+
+    it('should call onBlur with the current value', () => {
+        const blurFn = jest.fn();
+        const value = { hex: '#ccc', alpha: 1, isValid: true };
+        const component = mount(<ColorInput value={value} onBlur={blurFn} />);
+        component.find('input[type="text"]').simulate('blur');
+        expect(blurFn).toHaveBeenCalledWith(value);
+    });
+
+    it('should set the color sample according to the value', () => {
+        const value = { hex: '#ccc', alpha: 1, isValid: true };
+        const component = mount(<ColorInput value={value} />);
+        expect(component.find('ColorSample').prop('value')).toEqual(value);
+    });
+
+    it('should update the color sample on change', () => {
+        const value = { hex: '#ccc', alpha: 1, isValid: true };
+        const component = mount(<ColorInput value={value} />);
+        component.find('input[type="text"]').simulate('change', { target: { value: '#fff' } });
+        expect(component.find('ColorSample').prop('value')).toEqual({
```

Comment:
in component testing we should focus on the final result instead of internal details, for example instead of checking for props passed to internal component or checking states we should check for the color rendered, then we should find the HTML element that render this and check what color it have

Reply:
fixed in #2323